### PR TITLE
gh-107901: add the HAS_EVAL_BREAK instruction flag

### PR DESCRIPTION
**Pull request by iritkatriel**
This is motivated by https://github.com/python/cpython/issues/107901, which implies we need to make sure that all instructions the have an eval breaker check also have a line number. 

I'm not adding that assertion here because it will fail, so that's going to require more work. 

* Issue: gh-107901

### CODE REVIEW

File: Tools/cases_generator/flags.py [hunk 1]
```diff
@@ -35,13 +36,13 @@ def fromInstruction(instr: parsing.Node) -> "InstructionFlags":
             HAS_FREE_FLAG=has_free,
             HAS_LOCAL_FLAG=(
                 variable_used(instr, "GETLOCAL") or variable_used(instr, "SETLOCAL")
-            )
-            and not has_free,
+            ) and not has_free,
```
Comment (iritkatriel) [inline Tools/cases_generator/flags.py]:
```suggestion
            )
            and not has_free,
```

File: Tools/cases_generator/flags.py [hunk 2]
```diff
@@ -53,10 +55,11 @@ def names(self, value: bool | None = None) -> list[str]:
             return list(dataclasses.asdict(self).keys())
         return [n for n, v in dataclasses.asdict(self).items() if v == value]
 
-    def bitmap(self) -> int:
+    def bitmap(self, ignore: list[str] = []) -> int:
```
Comment (gvanrossum) [inline Tools/cases_generator/flags.py]: Using a mutable default sets alarm bells off in my head, even though you're safe in this case. Maybe make it an abstract set?

**Review by gvanrossum — COMMENTED** [review]
I like how the flags framework is so easily extensible.

File: Tools/cases_generator/analysis.py [hunk 3]
```diff
@@ -381,7 +381,8 @@ def analyze_pseudo(self, pseudo: parsing.Pseudo) -> PseudoInstruction:
         # Make sure the targets have the same fmt
         fmts = list(set([t.instr_fmt for t in targets]))
         assert len(fmts) == 1
-        assert len(list(set([t.instr_flags.bitmap() for t in targets]))) == 1
+        ignored_flags = set(('HAS_EVAL_BREAK_FLAG',))
+        assert len(list(set([t.instr_flags.bitmap(ignore=ignored_flags) for t in targets]))) == 1
```
Comment (gvanrossum) [inline Tools/cases_generator/analysis.py]: Use set and set comprehension notation:
```suggestion
        ignored_flags = {'HAS_EVAL_BREAK_FLAG'}
        assert len({t.instr_flags.bitmap(ignore=ignored_flags) for t in targets}) == 1
```

File: Tools/cases_generator/flags.py [hunk 4]
```diff
@@ -53,10 +56,11 @@ def names(self, value: bool | None = None) -> list[str]:
             return list(dataclasses.asdict(self).keys())
         return [n for n, v in dataclasses.asdict(self).items() if v == value]
 
-    def bitmap(self) -> int:
+    def bitmap(self, ignore: AbstractSet[str] = set()) -> int:
```
Comment (gvanrossum) [inline Tools/cases_generator/flags.py]:
```suggestion
    def bitmap(self, ignore: AbstractSet[str] = frozenset()) -> int:
```

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**